scipy/interpolate/src/_fitpackmodule.c

```diff
@@ -192,26 +243,24 @@ fitpack_bispev(PyObject *dummy, PyObject *args)
     ny = PyArray_DIMS(ap_ty)[0];
     mx = PyArray_DIMS(ap_x)[0];
     my = PyArray_DIMS(ap_y)[0];
-    /* Limit is min of (largest array size, max of Fortran int) */
-    int_max = (F_INT_MAX < NPY_MAX_INTP) ? F_INT_MAX : NPY_MAX_INTP;
-    /* v = int_max/my is largest integer multiple of `my` such that
-       v * my <= int_max
-    */
-    if (my != 0 && int_max/my < mx) {
-        /* Integer overflow */
-        PyErr_Format(PyExc_RuntimeError,
-                     "Cannot produce output of size %dx%d (size too large)",
-                     mx, my);
+
+    mxy = _mul_overflow_intp(mx, my);
+    if (mxy < 0) {
         goto fail;
     }
-    mxy = (npy_intp)mx * (npy_intp)my;
+
     ap_z = (PyArrayObject *)PyArray_SimpleNew(1,&mxy,NPY_DOUBLE);
     if (ap_z == NULL) {
         goto fail;
     }
     z = (double *) PyArray_DATA(ap_z);
     if (nux || nuy) {
-        lwrk = mx*(kx + 1 - nux) + my*(ky + 1 - nuy) + (nx - kx - 1)*(ny - ky - 1);
+        /* lwrk = mx*(kx + 1 - nux) + my*(ky + 1 - nuy) + (nx - kx - 1)*(ny - ky - 1); */
+        lwrk = _mul_overflow_f_int(nx - kx - 1, ny - ky - 1);
+        if (lwrk < 0) {
+            goto fail;
+        }    
+        lwrk += mx*(kx + 1 - nux) + my*(ky + 1 - nuy);
```

Comment:
If the m-side (rather than the n-side) is unlikely to overflow, can we have a comment why we're not bothering with the overflow check here?

Or maybe do something like (here I'm adding a basic check that the sum doesn't overflow either; not sure if the summands involved can ever be substantially negative, in which case this would need more finesse):
```suggestion
        // lwrk = mx*(kx + 1 - nux) + my*(ky + 1 - nuy) + (nx - kx - 1)*(ny - ky - 1)
        prod1 = _mul_overflow_f_int(mx, kx + 1 - nux);
        prod2 = _mul_overflow_f_int(my, ky + 1 - nuy);
        prod3 = _mul_overflow_f_int(nx - kx - 1, ny - ky - 1);
        lwrk = prod1 + prod2 + prod3
        // if the sum becomes smaller than one of the (positive?!?) summands, we overflowed
        if (prod1 < 0 || prod2 < 0 || prod3 < 0 || lwrk < prod1 || lwrk < prod2 || lwrk < prod3) {
            goto fail;
        }
```

Reply:
kx and ky are at most 5, while m* and n* are input data size

Reply:
`n*` extends to `nux` as well? I mean, if we clean up the overflow handling, we could do it without leaving that margin of error, but as I said, even just a comment why we're not bothering with overflow checking for the other product / sum would be already sufficient.

Reply:
Actually, nu{x, y} are  user requested derivatives, so positive integers from 0 to 5. The main danger is `m{x, y}` which are data size.

That said, let's just roll with your suggestion, which is now included in the PR

Reply:
> That said, let's just roll with your suggestion, which is now included in the PR

Thanks, though it was more of a draft suggestion and not really intended for a direct commit. E.g. I wouldn't expect the `(positive?!?)` to stay in the code -- we should think about whether they're always positive or not and adapt the code as necessary (or fall back to just adding a comment, I'm not trying to make your life difficult 😅)

Reply:
How about we roll back your (draft) suggestion, keep the original partial guard, add a code comment and work through additional issues as they show up in real usage?

Reply:
Fine by me 👍

Reply:
Ok, done in https://github.com/scipy/scipy/pull/17212/commits/fcdc0f1bae69c4d59b8703fdbc325c4543dae427